### PR TITLE
Table column

### PR DESCRIPTION
**Pull request by joakim-hove**
This PR is a followup to: https://github.com/OPM/opm-parser/pull/649

The initial purpose of this PR was to adapt to the new Table api from opm-parser, however as a side effect the main volume in changes is in the `flow_solvent` and `flow_polymer` implementations where the `Deck` class was incorrectly used to infer solvent and polymer properties respectively.

OK - this has now run through Norne successfully.

### CODE REVIEW

**Comment by atgeirr** [comment]
> NB: There is currently a failing test in here!

All tests pass for me in opm-autodiff, did you mean in this PR?

**Comment by joakim-hove** [comment]
> All tests pass for me in opm-autodiff, did you mean in this PR?

I mean this PR - working on it.

**Comment by GitPaean** [comment]
Strange. Only one of the tests in opm-core failed. The tests from opm-autodiff all passed from my side in debug mode.

BTW, one simple running indicates the PR did not change the simulation results of flow_polymer.

**Comment by joakim-hove** [comment]
> The tests from opm-autodiff all passed from my side in debug mode.

You have run this PR? The test which fails is the `flow_SPE1` test - that will only run if `opm-data` has been found in the configure process.

**Comment by GitPaean** [comment]
Yes. I did not integrate the tests in `opm-data`.

**Comment by GitPaean** [comment]
And also, I tried to put `opm-data` in the same folder with `opm-autodiff` and `opm-parser` and other modules. OPM did not find `opm-data` during the configure process. 

Could you tell a little bit about your OPM folder structure?

**Comment by GitPaean** [comment]
I found the problem. The problem is that I use soft link to the `opm-data` directory. And the other modules are also soft links, while the actual folder of `opm-data` is not in the same directory with `opm-parser` and etc.

**Comment by GitPaean** [comment]
With a rebuilding, 
```
Start 13: flow_SPE1
13/13 Test #13: flow_SPE1 ........................   Passed    2.89 sec
```
I did not reproduce your problem. I do not know what the reason is.

**Comment by joakim-hove** [comment]
Ok - thank you very much for the effort.

Actually I had å mixture of debug and release build, maybe that was the
problem.
6. jan. 2016 16:30 skrev "Kai Bao" <notifications@github.com> følgende:

**Comment by joakim-hove** [comment]
> The problem is that I use soft link to the opm-data 

OK - I know @atgeirr fixed some exotic issues related to symlinks in opm-common some time ago, maybe if you update your opm-common distribution?

**Comment by GitPaean** [comment]
My `opm-common` is updated. Maybe I can talk with him about this when we get a chance later.

**Comment by joakim-hove** [comment]
Yes - now it works for me too!

File: opm/polymer/PolymerInflow.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ namespace Opm
 
     /// Constructor.
     /// @param[in]  deck     Input deck expected to contain WPOLYMER.
-    PolymerInflowFromDeck::PolymerInflowFromDeck(Opm::DeckConstPtr deck,
+    /*PolymerInflowFromDeck::PolymerInflowFromDeck(Opm::DeckConstPtr deck,
```
Comment (atgeirr) [inline opm/polymer/PolymerInflow.cpp]: This constructor can be deleted.

File: opm/polymer/PolymerInflow.hpp [hunk 2]
```diff
@@ -92,17 +92,16 @@ namespace Opm
         /// \param[in]  deck        Input deck expected to contain WPOLYMER.
         /// \param[in]  wells       Wells structure.
         /// \param[in]  num_cells   Number of cells in grid.
-        PolymerInflowFromDeck(Opm::DeckConstPtr deck,
-                              const Wells& wells,
-                              const int num_cells);
+        //PolymerInflowFromDeck(Opm::DeckConstPtr deck,
```
Comment (atgeirr) [inline opm/polymer/PolymerInflow.hpp]: Remove this declaration.

File: opm/polymer/PolymerProperties.hpp [hunk 3]
```diff
@@ -147,14 +147,14 @@ namespace Opm
             const auto& plyviscTable = tables->getPlyviscTables().getTable<PlyviscTable>(0);
 
 
-            c_vals_visc_ = plyviscTable.getPolymerConcentrationColumn();
-            visc_mult_vals_ =  plyviscTable.getViscosityMultiplierColumn();
+            c_vals_visc_ = plyviscTable.getPolymerConcentrationColumn().vectorCopy( );
```
Comment (atgeirr) [inline opm/polymer/PolymerProperties.hpp]: Why is `getNamedColumn().vectorCopy()` used some places and `exportColumn("NAME_STRING")` in others?

File: opm/polymer/fullyimplicit/SimulatorFullyImplicitBlackoilPolymer_impl.hpp [hunk 4]
```diff
@@ -110,7 +110,7 @@ namespace Opm
             if (wells_manager.c_wells() == 0) {
                 OPM_THROW(std::runtime_error, "Cannot control polymer injection via WPOLYMER without wells.");
             }
-            polymer_inflow_ptr.reset(new PolymerInflowFromDeck(deck_, BaseType::eclipse_state_, *wells, Opm::UgGridHelpers::numCells(BaseType::grid_), timer.currentStepNum()));
+            polymer_inflow_ptr.reset(new PolymerInflowFromDeck(BaseType::eclipse_state_, *wells, Opm::UgGridHelpers::numCells(BaseType::grid_), timer.currentStepNum()));
```
Comment (atgeirr) [inline opm/polymer/fullyimplicit/SimulatorFullyImplicitBlackoilPolymer_impl.hpp]: Is there a way to ask the eclipse state if there are polymer wells? If so we could get rid of the deck_ member in this class (now just used on line 109). If there is no such thing, do not bother, it's not a very big deal.

File: opm/polymer/PolymerInflow.hpp (hunk 2 above)
Comment (joakim-hove) [inline opm/polymer/PolymerInflow.hpp]: Fixup pushed.

**Comment by atgeirr** [comment]
I think this is ready to be merged (and the PR in opm-core as well), so feel free to press green when you'd like @joakim-hove.

Thanks for the effort!

PS. I still wonder about vectorCopy() vs. exportColumn() (but that is not a reason to avoid merging I think).

File: opm/polymer/PolymerProperties.hpp (hunk 3 above)
Comment (joakim-hove) [inline opm/polymer/PolymerProperties.hpp]: Change of opinion; currently (I think) the state of the matter is that `exportColumn(  )` is used in opm-core and `vectorCopy( )` is used in opm-autodiff. As mentioned in the opm-core PR I would like to go to JSON based support of the column names. I have not been able to determine for myself which of the two is the best intermediate stage.

I would happy to unify though.

File: opm/polymer/fullyimplicit/SimulatorFullyImplicitBlackoilPolymer_impl.hpp (hunk 4 above)
Comment (joakim-hove) [inline opm/polymer/fullyimplicit/SimulatorFullyImplicitBlackoilPolymer_impl.hpp]: Currently not :-(

**Comment by joakim-hove** [comment]
> so feel free to press green when you'd like

Thank you!

File: opm/polymer/PolymerProperties.hpp (hunk 3 above)
Comment (atgeirr) [inline opm/polymer/PolymerProperties.hpp]: No need, just curious. Fixate on one if you eventually decide one of them is better.

**Pull request merged**